### ISSUE TITLE
Markdown Preview clickable links to headings (ToC links, for example)

### ISSUE THREAD

**Issue by guydavies** [zed-industries/zed#18699]
### Check for existing issues

- [X] Completed

### Describe the feature

In the Markdown preview, links to external files are clickable and open that file.  I would like to have a similar capability for links to headings using the [label](#go-to-this-heading) format.  This is possible in other editors (the obvious one being vscode with suitable extensions).

Ideally, clicking the link in the preview would scroll up/down to the matching heading/anchor in the preview.  Even better would be if the editor and preview panes were synced so the editor also scrolled to the relevant point (probably another feature request).

**Comment by skejeton** [comment]
Agreed, I might implement a pull request for this feature in my free time, as I utilize tables of content often for internal documents.

**Comment by erikhsnr** [comment]
it would be amazing if the `[TOC]` command works aswell then to create a Table of Contents.

**Comment by nic-obert** [comment]
Any updates on this issue? This feature is fundamental when navigating large documentation files. Paired with the inability to search markdown previews with `ctrl-f` described in #24273, it makes for a bad experience when working with markdown files.

**Comment by lukejans** [comment]
To add to this, I think it would be great to take the [GitHub-flavoured approach to linking to headings](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#section-links). GitHub calls these "section links" which use the following syntax: `[description](./folder/file.md#heading)`. The link provided above goes into more details on how GitHub handles whitespace and special characters for headings.

I noticed that the outline panel already implements the link feature to go to a specific heading. Correct me if I'm wrong but the only feature that would need some real work would be the preview?
